test/common/http/async_client_impl_test.cc

```diff
@@ -544,6 +550,7 @@ TEST_F(AsyncClientImplTest, ResetInOnHeaders) {
       .WillOnce(Invoke([&stream](HeaderMap&, bool) { stream->reset(); }));
   EXPECT_CALL(stream_callbacks_, onData(_, _)).Times(0);
   EXPECT_CALL(stream_callbacks_, onReset());
+  EXPECT_CALL(stream_callbacks_, onComplete());
```

Comment:
I can't seem to reason why onComplete is being called here. Afaict local should not be closed, and thus onComplete should not happen. Need to think about this in the morning.

Reply:
I don't think we can merge until we understand this. Only one or the other should ever be called.

Reply:
resolved, see comments.